### DISCUSSION TITLE
Pre-process a file to generate a renovate config

### DISCUSSION THREAD

**Discussion by nitrocode** [renovatebot/renovate#30563] [Suggest an Idea]
### Tell us more.

I understand that you folks do not want to support yaml as a config format (re: https://github.com/renovatebot/renovate/issues/7031)

What if we, instead, could generate a renovate json or json5 file prior to renovate running ?

This way, we could maintain our config in yaml, run a command to convert it, and then get an outputted json/json5 that renovate can read.

**Comment by rarkins** [comment]
Global or repo config?

**Reply by nitrocode** [reply]
Both would be nice 😁
